zerver/lib/actions.py

```diff
@@ -4365,7 +4365,36 @@ def user_info(um: UserMessage) -> Dict[str, Any]:
             'id': um.user_profile_id,
             'flags': um.flags_list()
         }
-    send_event(user_profile.realm, event, list(map(user_info, ums)))
+
+    def subscriber_info(user_id: int) -> Dict[str, Any]:
+        return {
+            'id': user_id,
+            'flags': ['read']
+        }
+
+    subscribers_to_be_notified_ids = set()  # type: Set[int]
+    if message.is_stream_message():
+        thread_being_edited = Stream.objects.get(id=stream_id)
+
+        if thread_being_edited.is_history_public_to_subscribers:
+            subscribers = get_active_subscriptions_for_stream_id(stream_id)
+            # All users that are subscribed to the stream must be notified when a message is edited
+            subscribers_to_be_notified_ids = set([user.user_profile_id for user in subscribers])
+
+    users_to_be_notified = list(map(user_info, ums))
+    # Remove duplicates by excluding the id of users already in users_to_be_notified list.
+    # This is the case where a user has a UserMessage row and is a Subscriber
+    subscribers_to_be_notified_ids -= set([user["id"] for user in users_to_be_notified])
```

Comment:
I think you can replace this set subtraction stuff with using a `.exclude(user_profile__in=[um.in for um in ums])` in the `is_history_public_to_subscribers` block, which might be cleaner and faster.